operators/src/pro/ml/xgboost.rs

```diff
@@ -0,0 +1,694 @@
+        };
+
+        let init_rasters = future::try_join_all(
+            self.sources
+                .rasters
+                .iter()
+                .map(|raster| raster.clone().initialize(context)),
+        )
+        .await?;
+
+        let input = init_rasters.get(0).context(self::error::NoInputData)?;
+
+        let spatial_reference = input.result_descriptor().spatial_reference;
+
+        let in_descriptors = init_rasters
+            .iter()
+            .map(InitializedRasterOperator::result_descriptor)
+            .collect::<Vec<_>>();
+
+        for other_spatial_reference in in_descriptors.iter().skip(1).map(|rd| rd.spatial_reference)
+        {
+            ensure!(
+                spatial_reference == other_spatial_reference,
+                crate::error::InvalidSpatialReference {
+                    expected: spatial_reference,
+                    found: other_spatial_reference,
+                }
+            );
+        }
+
+        let time = time_interval_extent(in_descriptors.iter().map(|d| d.time));
+        let bbox = partitions_extent(in_descriptors.iter().map(|d| d.bbox));
+
+        let resolution = in_descriptors
+            .iter()
+            .map(|d| d.resolution)
+            .reduce(|a, b| match (a, b) {
+                (Some(a), Some(b)) => {
+                    Some(SpatialResolution::new_unchecked(a.x.min(b.x), a.y.min(b.y)))
+                }
+                _ => None,
+            })
+            .flatten();
+
+        let out_desc = RasterResultDescriptor {
+            data_type: RasterOut,
+            time,
+            bbox,
+            resolution,
+            spatial_reference,
+
+            // where do we get this information from?
+            // test_data/dataset_defs? If so, how to load it?
+            measurement: Measurement::Classification(ClassificationMeasurement {
+                measurement: "raw".into(),
+                classes: hashmap!(0 => "Water Bodies".to_string()),
+            }),
+        };
+        let initialized_operator = InitializedXgboostOperator {
+            result_descriptor: out_desc,
+            sources: init_rasters,
+            model_sub_path: model_path.expect("The modelfile should be there!"),
+            no_data_value: self.params.no_data_value,
+        };
+
+        Ok(initialized_operator.boxed())
+    }
+
+    span_fn!(XgboostOperator);
+}
+
+impl InitializedRasterOperator for InitializedXgboostOperator {
+    fn result_descriptor(&self) -> &RasterResultDescriptor {
+        &self.result_descriptor
+    }
+
+    fn query_processor(&self) -> Result<TypedRasterQueryProcessor> {
+        let vec_of_rqps: Vec<Box<dyn RasterQueryProcessor<RasterType = f32>>> = self
+            .sources
+            .iter()
+            .map(|init_raster| {
+                let typed_raster_qp = match init_raster.query_processor() {
+                    Ok(r) => r,
+                    Err(e) => {
+                        return Err(Error::XGBoost {
```

Comment:
why is an XGBoost error if the input raster processor could not be created? It has nothing to do with your operaror.

Also: you can use the `map_err` function instead of a match block if you want to replace the error of a result.